gbpservice/neutron/tests/unit/nfp/configurator/drivers/vpn/vyos/test_vpn_driver.py

```diff
@@ -0,0 +1,424 @@
+        self.plugin_rpc = vpn.VpnaasRpcSender(self.context,
+                                              self.dict_objects.sc)
+        self.driver = vyos_vpn_driver.VpnaasIpsecDriver(self.plugin_rpc)
+        self.svc_validate = vyos_vpn_driver.VPNSvcValidator(self.plugin_rpc)
+        self.resp = mock.Mock()
+        self.fake_resp_dict = {'status': True}
+
+    def test_create_vpn_service(self):
+        '''
+        Implements method to test the vpn driver's create vpn service.
+        '''
+
+        context = self.dict_objects._make_service_context(operation_type='vpn')
+
+        kwargs = self.dict_objects._make_kwargs(operation='create',
+                                                service_type='vpn')
+        with mock.patch.object(self.plugin_rpc, 'update_status') as (
+                                                mock_update_status):
+            self.driver.vpnservice_updated(context, kwargs)
+            mock_update_status.assert_called_with(
+                                        self.context,
+                                        self.dict_objects.vpn_vpnsvc_active)
+
+    def test_create_ipsec_site_conn(self):
+        '''
+        Implements method to test the vpn driver's create ipsec site conn
+        '''
+        self.resp = mock.Mock(status_code=200)
+        context = self.dict_objects._make_service_context()
+        kwargs = self.dict_objects._make_kwargs(operation='create',
+                                                service_type='ipsec')
+        with mock.patch.object(self.plugin_rpc, 'update_status') as (
+                                                mock_update_status),\
+            mock.patch.object(json, 'loads') as mock_resp,\
+            mock.patch.object(requests, 'post') as (
+                                                mock_post):
+            mock_resp.return_value = self.fake_resp_dict
+            mock_post.return_value = self.resp
+            self.driver.vpnservice_updated(context, kwargs)
+
+            mock_post.assert_called_with(
+                            self.dict_objects.url_create_ipsec_conn,
+                            data=self.dict_objects.ipsec_data,
+                            timeout=self.dict_objects.timeout)
+            mock_update_status.assert_called_with(
+                                        context,
+                                        self.dict_objects.ipsec_vpnsvc_status)
+
+    def test_delete_ipsec_site_conn(self):
+        '''
+        Implements method to test the vpn driver's create ipsec site conn
+        '''
+
+        self.resp = mock.Mock(status_code=200)
+        kwargs = self.dict_objects._make_kwargs(operation='delete',
+                                                service_type='ipsec')
+        with mock.patch.object(self.plugin_rpc, 'update_status'),\
+            mock.patch.object(json, 'loads') as mock_resp,\
+            mock.patch.object(requests, 'delete') as (
```

Comment:
Data is passed along with delete request. We need to mock the json dumps method also in case where data is passed.